src/config/session.c

```diff
@@ -101,16 +104,24 @@ update_activation_env(const char *env_keys)
 	const char *systemd = "systemctl --user import-environment ";
 
 	cmd = calloc(1, strlen(dbus) + strlen(env_keys) + 1);
-	strcat(cmd, dbus);
-	strcat(cmd, env_keys);
-	spawn_async_no_shell(cmd);
-	free(cmd);
+	if (!cmd) {
+		wlr_log(WLR_ERROR, "Failed to allocate memory for dbus env update");
```

Comment:
IMO it's makes sense do it like this:

```
cmd = calloc(...);
if (!cmd) {
        oom(); // writes to log and exit(1)
}
```

or simply

```
cmd = xcalloc(...); // exits on allocation error
````

Reply:
As labwc is a compositor it will take down everything else (at least graphical applications) with it. But honestly if we can't even allocate a few bytes for a string lots of other places will break as well (likely including the `wlr_log()` statement).

Reply:
Exactly. It's better to exit early than to start with half-baked labwc.